tools/awslint/lib/rules/construct.ts

```diff
@@ -0,0 +1,46 @@
+import reflect = require('jsii-reflect');
+import { Linter, MethodSignatureParameterExpectation } from '../linter';
+import { CONSTRUCT_FQN, isConstruct } from '../util';
+
+interface ConstructLinterContext  {
+  readonly construct: reflect.ClassType;
+}
+
+export const constructLinter = new Linter<ConstructLinterContext>(assembly => assembly.classes
+  .filter(t => isConstruct(t))
+  .map(construct => ({ construct })));
+
+constructLinter.add({
+  code: 'construct-ctor',
+  message: 'signature of all construct constructors should be "scope, id, props"',
```

Comment:
Only of non-abstract classes I'd think. That gives the ability to have helper abstract classes take some additional construct args.

Reply:
Good call. I'll fix.